File: algorithms/python/7.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x==0:
            return x
        sign = 1 if x > 0 else -1
        x = abs(x)
        res = 0
        while x:
            res *= 10
            res += x%10
            x /= 10
        if res > 2**31 + 1 or res < -2**32 - 1:
            return 0
        return res * sign
        

File: algorithms/cpp/7.cpp
class Solution {
public:
    int reverse(int x) {
        if (0 == x)
        {
            return 0;
        }
        long long int res = 0;
        int sign = x > 0 ? 1:-1;
        x = abs(x);
        while(x)
        {
            res *= 10;
            res += x%10;
            x /= 10;
        }
        if((res > INT_MAX)  || (res < INT_MIN))
        {
            return 0;
        }
        return res * sign;
    }
};